### DISCUSSION TITLE
Toggle for Group Mode?

### DISCUSSION THREAD

**Discussion by cmeid** [maca88/SmartBikeLights#38] [Q&A]
I've said it before, but I love this software. It just tickles my little nerd heart in a wonderful way.

Quick question: I mostly use SBL to leave the Varia off when it doesn't detect anything, and turn it on when either a threat is detected or I brake. However, I frequently ride both with groups and alone. I When with a group, I would like to always only turn on Peloton mode. When I am alone, I want to use a combination of day/night/solid that I have configured.

I understand that I can use primary and secondary configurations for this, but I would like to be able to change the mode more easily - ideally just from the "1 Field Layout" data screen. I ride gravel with a fast group, and if I forget to change the mode then I need to be able to do it very quickly when someone complains that my taillight is blinding them, and I don't have time to go into the app config.

Can anyone think of a way that this might be done using the configuration options available? I was wondering if maybe there is a configuration in the light panel setting that I can abuse somewhat.

Ideally, I would like to have a configurable "mode switch" on the data screen view, and then use the setting of that switch as a filter in my configuration.

Here is my current configuration for reference:

Solo Mode - Primary (Non-Peloton Light Mode)
###0,73535488::1#6,4!DFlash:2:7:30:0I]-1]0Er0,s0!NFlash:2:6:30:0I]-1]0Es0,r0!Brake:1:4:5:0A[-10!:1:0:0:0D=1##5,3:RTL515:0:16777215!2,:-1,Off:0!1,High:4!2,Day Flash:7,Night Flash:6#0::#0:0#123!:123!#0#B3843##3#0#0

Group Mode - Secondary (Only Peloton Light Mode)
###0,73535488::1#3,3!Threat:1:5:30:0I]-1]0!Brake:1:5:5:0A[-10!:1:0:0:0D=1##5,3:RTL515:0:16777215!2,:-1,Off:0!1,High:4!2,Day Flash:7,Night Flash:6#0::#0:0#123!:123!#0#B3843##3#0#0

**Comment by cmeid** [comment]
Update - re-reading the documentation, I think the built in configuration editor will work for this purpose: great!

Having a toggle on the data field view would be even nicer, but this is a fine solution.

That does leave me with a feature request @maca88: it would be amazing to have a name for the configuration, and display that on the data field. Then I could easily see and verify whether I am in "Solo Mode" or "Group Mode"

Thanks again for this great software!

**Comment by maca88** [comment]
Hi,

Now with version `0.25.0` it is possible to set the name for each configuration:
![image](https://user-images.githubusercontent.com/5311349/221390250-4228c0de-940c-420b-9118-77eabd021267.png)

Also, you can add a button on the light panel to switch between them:
![image](https://user-images.githubusercontent.com/5311349/221390277-4407c869-e219-4fa2-bef5-180876b4371f.png)

This is how it would look like for your case:
![image](https://user-images.githubusercontent.com/5311349/221390301-225927bb-80e6-4d59-80bd-189fea739d75.png)

Give it a try and let me know if you have any issues with it.

**Reply by cmeid** [reply]
Damn, you rock! I wish the big software vendors were this responsive! Are you sure there's not a Kaffeekasse (oder ne Matekasse) somewhere to sponsor your coding?

I have just prepared my configuration, I will test it on our group ride tomorrow night here in Berlin and report back.

Thank you!

**Reply by maca88** [reply]
Currently, I haven't set up a Kaffeekasse yet. I may do that someday. Thanks anyways :)
